### DISCUSSION TITLE
Editor become inoperable when open a compressed javascript file

### DISCUSSION THREAD

**Discussion by zhaomenghuan** [eclipse-theia/theia#11391] [Improvements] [answered]
When I open a compressed javascript file in IDE based on Theia, editor becomes inoperable，e.g. can't scroll and no highlighting，but it can be opened very quickly in VS Code.There is still a big gap between Theia Editor editing experience and VS Code. It is recommended to optimize it specially.

**Comment by vince-fugnitto** [comment, marked as answer]
@zhaomenghuan thank you for the discussion, I believe it is related to `tokenization`, where if a line exceeds the character limit or if it takes too long to tokenize we should not tokenize it.

Do you mind trying the following commit and seeing if it works well for opening such documents: https://github.com/eclipse-theia/theia/commit/351de6a64db5bc409e693c8b518e85df9c7efa17.
